### PR TITLE
jQuery UI upgrade to 1.12

### LINKED ISSUE #103: Upgrading to v1.12.0
Please add this version to the gem. I'd like to use jQuery 3, which has a bug with v1.11.4 in the tooltip. This is fixed in v1.12.0

### PR DESCRIPTION
**Pull request by Borzik**
Since directory structure has changed in 1.12, previous build system didn't work for 1.12. 
jQuery UI doesn't have build info files, so there's no optimal way to get dependency list for each file. I ended up with parsing `define([...deps], factory)` block of each JS file.

As a result:
- we completely follow jQuery UI 1.12 folder structure (only `ui` subfolder has been dropped for usage simplicity)
- widgets, i18n and effects now have their own directories
- gem version went up to 6.0.0, since this is a breaking change for most users, and since we upgrade to 1.12
- **also I had to upgrade Travis Ruby version to 2.2.2 because 1.9.3 doesn't work with your gemspec anymore**

I've also updated readme.md to follow new directory structure.

Please let me know any questions you have.

And it closes #103

### CODE REVIEW

**Comment by seanfcarroll** [comment]
Please merge this :-)

**Comment by ttrmw** [comment]
+1

**Comment by Hunterlong** [comment]
+1

**Comment by okamos** [comment]
+1

**Comment by joliss** [comment]
Thanks for the PR! Some comments:

* I cannot seem to install the "json" gem with Ruby 2.3. I've had to update the dependency like so:

```ruby
diff --git a/jquery-ui-rails.gemspec b/jquery-ui-rails.gemspec
index 4ba6e19..43d58e9 100644
--- a/jquery-ui-rails.gemspec
+++ b/jquery-ui-rails.gemspec
@@ -15,7 +15,7 @@ Gem::Specification.new do |s|

   s.add_dependency "railties", ">= 3.2.16"

-  s.add_development_dependency "json", "~> 1.7"
+  s.add_development_dependency "json", "~> 2.0.2"

   s.files        = `git ls-files`.split("\n").reject { |f| f =~ /^testapp|^jquery-ui/ }
   s.executables  = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
```

* When I merge your PR and run `rake`, it leaves `app/assets/javascripts/jquery-ui/widgets/` empty and `git status` shows a bunch of files changed. I'd expect it to just regenerate everything in `app/assets` without change.

**Comment by joliss** [comment]
* Some of your commits are just fixes to earlier commits. Can you use `git rebase -i` and squash them accordingly, please?

**Comment by joliss** [comment]
> When I merge your PR and run `rake`, it leaves `app/assets/javascripts/jquery-ui/widgets/` empty and `git status` shows a bunch of files changed. I'd expect it to just regenerate everything in `app/assets` without change.

Never mind this, I forgot to run `git submodule update`.

File: app/assets/javascripts/jquery-ui.js [hunk 1]
```diff
@@ -1,37 +1,56 @@
...
-//= require jquery-ui/resizable
-//= require jquery-ui/selectable
-//= require jquery-ui/selectmenu
-//= require jquery-ui/slider
-//= require jquery-ui/sortable
-//= require jquery-ui/spinner
-//= require jquery-ui/tabs
-//= require jquery-ui/tooltip
-//= require jquery-ui/widget
+//= require jquery-ui/core.js
```
Comment (joliss) [inline app/assets/javascripts/jquery-ui.js]: I don't think there should be a trailing `.js` here, should there?

**Comment by Borzik** [comment]
@joliss thanks for review!
**Re: JSON gem:** initial gemspec worked for me when I tested everything, but anyway, I've updated it with the code you had provided.
**Re: trailing `.js` extensions:** I've missed those somehow. Pushing a fix.
**Re: commits that are fixes:** it seems like you can [squash all commits from PR together on merge](https://github.com/blog/2141-squash-your-commits). If you still think it would be better if I did that — I'll have to close this PR and create a new one from another branch, because the source branch is being used by me in some projects already, and someone else could have used that as well.

**Comment by joliss** [comment]
Great, thanks! I've released v6.0.0 of this gem.

**Pull request merged**